pac4j-core/src/main/java/org/pac4j/core/profile/Pac4JPrincipal.java

```diff
@@ -0,0 +1,43 @@
+package org.pac4j.core.profile;
+
+import java.security.Principal;
+import java.util.Objects;
+
+import org.pac4j.core.util.CommonHelper;
+
+public class Pac4JPrincipal implements Principal {
+
+    private final String id;
+
+    public Pac4JPrincipal(UserProfile profile) {
```

Comment:
Just wondering whether it would be better to have the identifier directly... as indeed, the whole `UserProfile` is not necessary... but at the same time, this makes this `Pac4JPrincipal` component fully tied to pac4j...

If we keep the `UserProfile` in the constructor, we should add a `CommonHelper.assertNotNull` on it...

Reply:
My idea was that the principal was derived from profile, so we pass the profile to it, and it is the responsibility of this implementation to take from it what it needs.
Like this we are sure that everybody using this implementation will use the same information from the profile to build the principal.

Also I am wondering if it should use `id` or `username` from the profile…

OK for the rest :)

Reply:
About the id/username, the question is whether the `getName()` must return a unique identifier or something more meaningful.

Reply:
I was more thinking about `getUsername()` since it is what represents user identity… there is some chance for `getId()` to be a db id or something like this…

Reply:
OK. Let's do that. And if the username is blank, let's use the identifier...

Reply:
makes sense!

Reply:
Of course, things are never easy :)
The `UserProfile` does not have a `getUsername()` method!

But I don't see any reason to work at the `CommonProfile` level, the Principal should be available for any profile retrieved from a provider.

So I will use `getId()` and we will see later if some other need arises.

Reply:
We never use the `UserProfile` which is abstract. Why not move things at the `CommonProfile` level?

Reply:
ok, that is an acceptable alternative, but why does UserProfile exist then?
I will use CommonProfile :)

Reply:
Its methods could have been merged into `CommonProfile`, but it's a design split: `UserProfile` contains generic user profile methods while `CommonProfile` contains common attributes methods.
As we want to enforce people to use common profiles, `UserProfile` is abstract and never really used.

Reply:
ok, make sense :)

so I pushed the last changes, I think you can validate and finally merge if I didn't forget anything :)